tests/console/journalctl.pm

```diff
@@ -7,21 +7,113 @@
     }
+    # Enable persistent journal and reboot system
+    assert_script_run("sed -i 's/.*Storage=.*/Storage=persistent/' /etc/systemd/journald.conf");
+    assert_script_run("sed -i 's/.*Seal=.*/Seal=yes/' /etc/systemd/journald.conf");
+    assert_script_run("systemctl restart systemd-journald");
+    # Setup FSS keys before reboot
+    assert_script_run('journalctl --interval=10s --setup-keys | tee journalctl-setup-keys.txt');
+    assert_script_run('journalctl --rotate');
+    assert_script_run("date '+%F %T' > reboottime");
+    assert_script_run("echo 'The batman is going to sleep' | systemd-cat -p info -t batman");
+    power_action('reboot', textmode => 1);
+    $self->wait_boot;
+    select_console 'root-console';
+    # Check journal state after reboot to trigger bsc#1171858
+    record_soft_failure "bsc#1171858" if (script_run('journalctl --verify --verify-key=`cat journalctl-setup-keys.txt`') != 0);
+    # Basic journalctl tests: Export journalctl with various arguments and ensure they are not empty
+    check_journal('',                      "journalctl.txt",        "journalctl empty");
+    check_journal('--boot=-1',             "journalctl-1.txt",      "journalctl of previous boot empty");
+    check_journal('-S "`cat reboottime`"', "journalctl-after.txt",  "journalctl after reboot empty");
+    check_journal('-U "`cat reboottime`"', "journalctl-before.txt", "journalctl before reboot empty");
+    check_journal("-k",                    "journalctl-dmesg.txt",  "journalctl dmesg empty");
+    assert_script_run('journalctl --identifier=batman --boot=-1| grep "The batman is going to sleep"', fail_message => "Error getting beacon from previous boot");
+    # Create virtual serial console for journal redirecting
+    script_run('socat pty,raw,echo=0,link=/dev/ttyS100 pty,raw,echo=0,link=/dev/ttyS101 & true');
+    assert_script_run('jobs | grep socat', fail_message => "socat is not running");
+    # Redirect journal to virtual serial console and syslog
+    assert_script_run('mkdir -p /etc/systemd/journald.conf.d/');
+    assert_script_run('echo -e "[Journal]\nForwardToConsole=yes\nTTYPath=/dev/ttyS100\nMaxLevelConsole=info" > /etc/systemd/journald.conf.d/fw-ttyS100.conf');
+    assert_script_run('echo -e "ForwardToSyslog=yes" >> /etc/systemd/journald.conf.d/fw-ttyS100.conf') if $check_syslog;
+    assert_script_run('systemctl restart systemd-journald.service');
+    script_run('cat /dev/ttyS101 > /var/tmp/journal_serial.out & true');
+    assert_script_run('echo "journal redirect output started (grep for: aeru4Poh eiDeik5l)" | systemd-cat -p info -t redirect');
+    # Write messages and check for them
+    assert_script_run('echo "We need to call batman" | systemd-cat -p info -t batman');
+    assert_script_run('echo "We NEED to call the batman NOW" | systemd-cat -p err -t batman');
+    assert_script_run("echo 'CALL THE BATMAN NOW!1!! AARRGGH!!' | systemd-cat -p emerg -t batman");
```

Comment:
XD

There is a typo:
```suggestion
    assert_script_run("echo 'CALL THE BATMAN NOW!!!! AARRGGH!!' | systemd-cat -p emerg -t batman");
```

Reply:
No typo, intentional over-exaggeration ;-)
https://www.urbandictionary.com/define.php?term=%21%21%211%2111%21%21%21oneoneeleven